calour/io.py

```diff
@@ -432,131 +513,105 @@ def read_open_ms(data_file, sample_metadata_file=None, gnps_file=None, feature_m
 
     Returns
     -------
-    Experiment
+    MS1Experiment
 
     See Also
     --------
     read
     '''
-    logger.debug('Reading OpenMS data (OpenMS bucket table %s, map file %s)' % (data_file, sample_metadata_file))
-    if rows_are_samples:
-        data_file_type = 'openms_transpose'
+    if data_file_type == 'mzmine2':
+        if sample_in_row is None:
```

Comment:
it is not like what you said. for example, in all `data_file_type` branches, when `sample_in_row` is `None`, it is set to `False` in all cases. so you can simplify this lengthy and confusing `if` blocks

Reply:
agree about sample_in_row, but i need to talk to priya again. i think the default for openms is sample_in_row=True, but not sure. If it is false, will change like you suggest.
regarding other variables, i don't think we have an option but setting them to None. could create a set_none internal function that covers the if to look more elegant. what do you think?

Reply:
fixed to passing a dict (data_table_params) which overrides the (ms file format dependent) defaults.